streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -77,10 +85,21 @@ abstract class KStreamKStreamJoin<K, VLeft, VRight, VOut, VThis, VOther> impleme
         this.enableSpuriousResultFix = windows.spuriousResultFixEnabled();
         this.joiner = joiner;
         this.outer = outer;
-        this.outerJoinWindowName = outerJoinWindowName;
+        this.outerJoinWindowStoreName = outerJoinWindowStoreFactory.map(StoreFactory::storeName);
+        this.outerJoinWindowStoreFactory = outerJoinWindowStoreFactory;
         this.sharedTimeTrackerSupplier = sharedTimeTrackerSupplier;
     }
 
+    @Override
+    public Set<StoreBuilder<?>> stores() {
+        final Set<StoreBuilder<?>> stores = new HashSet<>();
+        stores.add(new FactoryWrappingStoreBuilder<>(otherWindowStoreFactory));
+        if (outerJoinWindowStoreFactory.isPresent() && enableSpuriousResultFix) {
```

Comment:
Just clarifying: so far, as long as `spuriousResultFixEnabled` is true, the outer join should always present since it is just for this fix, right? Maybe worth adding a comment for future extensions if so.

Reply:
yes, that's true. I suspect maybe the `outerJoinWindowStoreFactory.isPresent()` check is just to avoid the warning about calling `outerJoinWindowStoreFactory.get()`?

But I'll add a comment

Reply:
Oh~ that's for checking, that's a bit surprising to me.